### PR TITLE
fix a bug in the code for new predicates.

### PR DESCRIPTION
**Pull request by restlessronin**
While attempting to add the new query predicates to node-tree-sitter ([see PR](https://github.com/tree-sitter/node-tree-sitter/pull/182)), based on the code in "lib/binding_web/binding.js", I created tests for the predicates. These tests reflect my understanding of the semantics for these predicates.

These tests do not pass until I fix what looks to be a bug in "bindings.js".  I have included the tests as well as the fixes in this PR.

If my understanding of the semantics is correct, the predicates will not work correctly until this PR is merged.

Update: I finally figured out how to build-wasm and run tests.

### CODE REVIEW

**Comment by restlessronin** [comment]
It would be great if the CI workflow is enabled, so we can see if the code works as is, and if not, the errors might provide enough information for me to fix the problem.

If the code works as is, and the tests are correctly written, then the new predicates are buggy and this PR should be merged quickly.

We are also waiting on this validation before merging [into node-tree-sitter](https://github.com/tree-sitter/node-tree-sitter/pull/182)

**Comment by restlessronin** [comment]
All the new tests are now succeeding locally. Please enable the CI to confirm this.

**Review by Amirmatinl — APPROVED** [review]

**Comment by restlessronin** [comment]
@Amirmatinl thanks for approving the PR. Just for completeness perhaps we can run the CI workflow so the PR can be merged.

**Comment by restlessronin** [comment]
Could the CI be enabled for the latest changes? Basically all the tests should be working now.

**Comment by restlessronin** [comment]
It would be good to get an approval for the CI workflow, so I know for sure that I don't have to do anything more at my end before the PR is reviewed.

**Comment by restlessronin** [comment]
Wondering if we can run the CI to validate (and hopefully close :-) this PR.

**Comment by restlessronin** [comment]
Looks like this is definitely a bug.

I hadn't seen [this earlier PR](https://github.com/tree-sitter/tree-sitter/pull/2795), which points out a similar issue, and seems to have additionally renamed (or correctly named) a predicate.

Would be good to get the tests validated by someone who can definitively say whether they are correct, and merge one of the PRs.

**Comment by amaanq** [comment]
Superseded by #2795, thank you for the PR though!

**Pull request closed without merging**
